extensions-contrib/opentelemetry-extensions/src/main/java/org/apache/druid/data/input/opentelemetry/protobuf/OpenTelemetryMetricsProtobufReader.java

```diff
@@ -0,0 +1,173 @@
+    }
+    catch (InvalidProtocolBufferException e) {
+      throw new ParseException(e, "Protobuf message could not be parsed");
+    }
+  }
+
+  private List<InputRow> parseMetricsData(final MetricsData metricsData)
+  {
+    return metricsData.getResourceMetricsList()
+        .stream()
+        .flatMap(resourceMetrics -> {
+          Map<String, Object> resourceAttributes = resourceMetrics.getResource()
+              .getAttributesList()
+              .stream()
+              .collect(Collectors.toMap(kv -> resourceAttributePrefix + kv.getKey(), kv -> kv.getValue().getStringValue()));
+          return resourceMetrics.getInstrumentationLibraryMetricsList()
+              .stream()
+              .flatMap(libraryMetrics -> libraryMetrics.getMetricsList()
+                  .stream()
+                  .flatMap(metric -> parseMetric(metric, resourceAttributes).stream()));
+        })
+        .collect(Collectors.toList());
+  }
+
+  private List<InputRow> parseMetric(Metric metric, Map<String, Object> resourceAttributes)
+  {
+    final List<InputRow> inputRows;
+    String metricName = metric.getName();
+    switch (metric.getDataCase()) {
+      case SUM: {
+        inputRows = new ArrayList<>(metric.getSum().getDataPointsCount());
+        metric.getSum()
+            .getDataPointsList()
+            .forEach(dataPoint -> inputRows.add(parseNumberDataPoint(dataPoint, resourceAttributes, metricName)));
+        break;
+      }
+      case GAUGE: {
+        inputRows = new ArrayList<>(metric.getGauge().getDataPointsCount());
+        metric.getGauge()
+            .getDataPointsList()
+            .forEach(dataPoint -> inputRows.add(parseNumberDataPoint(dataPoint, resourceAttributes, metricName)));
+        break;
+      }
+      // TODO Support HISTOGRAM and SUMMARY metrics
+      default:
+        throw new IllegalStateException("Unexpected value: " + metric.getDataCase());
+    }
+    return inputRows;
+  }
+
+  private InputRow parseNumberDataPoint(NumberDataPoint dataPoint,
+                                        Map<String, Object> resourceAttributes,
+                                        String metricName)
+  {
+
+    int capacity = resourceAttributes.size()
+            + dataPoint.getAttributesCount()
+            + 2; // metric name + value columns
+    Map<String, Object> event = Maps.newHashMapWithExpectedSize(capacity);
+    event.put(metricDimension, metricName);
+
+    if (dataPoint.hasAsInt()) {
+      event.put(valueDimension, dataPoint.getAsInt());
+    } else if (dataPoint.hasAsDouble()) {
+      event.put(valueDimension, dataPoint.getAsDouble());
+    } else {
+      throw new IllegalStateException("Unexpected dataPoint value type. Expected Int or Double");
+    }
+
+    event.putAll(resourceAttributes);
+    dataPoint.getAttributesList().forEach(att -> event.put(metricAttributePrefix + att.getKey(),
+        att.getValue().getStringValue()));
```

Comment:
same here, what happens if it's not a string value?

Reply:
Good point, I added a static method that checks `AnyValue` to make sure that they are only stringvalues for both resource and metric attributes